File: src/main/java/me/oriharel/machinery/fuel/FuelManager.java
package me.oriharel.machinery.fuel;

import com.google.gson.Gson;
import me.oriharel.machinery.Machinery;
import org.bukkit.Material;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public class FuelManager {

    private Machinery machinery;
    private Gson gson;
    private Material defaultType;
    private List<String> defaultLore;
    private String defaultDisplayName;

    public FuelManager(Machinery machinery) {
        this.machinery = machinery;
        this.gson = new Gson();
        YamlConfiguration configLoad = machinery.getFileManager().getConfig("fuel.yml").get();
        this.defaultType = Material.getMaterial(configLoad.getString("default_fuel_type"));
        this.defaultLore = configLoad.getStringList("default_lore");
        this.defaultDisplayName = configLoad.getString("default_display_name");

    }

    public Fuel getFuel(int amount, int energy) {
        return new Fuel(defaultType, energy, amount, defaultDisplayName, defaultLore);
    }

    public Fuel getFuel(Material material, int amount, int energy) {
        return new Fuel(material, energy, amount, defaultDisplayName, defaultLore);
    }

    public Fuel getFuel(ItemStack itemStack) {
        return new Fuel(itemStack, defaultLore);
    }

    public Fuel getFuel(ItemStack itemStack, int energy) {
        return new Fuel(itemStack, energy, defaultLore);
    }
}

File: src/main/java/me/oriharel/machinery/fuel/FuelManager.kt
package me.oriharel.machinery.fuel

import com.google.gson.Gson
import me.oriharel.machinery.Machinery
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class FuelManager(private val machinery: Machinery) {
    private val gson: Gson
    private val defaultType: Material?
    private val defaultLore: List<String>
    private val defaultDisplayName: String
    fun getFuel(amount: Int, energy: Int): Fuel {
        return Fuel(defaultType, energy, amount, defaultDisplayName, defaultLore)
    }

    fun getFuel(material: Material?, amount: Int, energy: Int): Fuel {
        return Fuel(material, energy, amount, defaultDisplayName, defaultLore)
    }

    fun getFuel(itemStack: ItemStack?): Fuel {
        return Fuel(itemStack, defaultLore)
    }

    fun getFuel(itemStack: ItemStack?, energy: Int): Fuel {
        return Fuel(itemStack, energy, defaultLore)
    }

    init {
        gson = Gson()
        val configLoad = machinery.fileManager.getConfig("fuel.yml").get()
        defaultType = Material.getMaterial(configLoad!!.getString("default_fuel_type")!!)
        defaultLore = configLoad.getStringList("default_lore")
        defaultDisplayName = configLoad.getString("default_display_name")!!
    }
}